File: lib/Logger.php
<?php

/**
 * Created by PhpStorm.
 * User: mrnda
 * Date: 19/02/2017
 * Time: 19:19
 */
class Logger
{
    const IS_DEBUG = true;

    public static function LogDebug($message)
    {
        if(IS_DEBUG)
        {
            printf("D: %s\n", $message);
        }
    }

    public static function LogError($message)
    {
        fprintf(STDERR, "E: %s\n", $message);
    }

    public static function printHelp()
    {
    }


}

File: src/logger.c
//
// Created by Michal Mrnustik on 28/02/2017.
//

#include "logger.h"


void logInfo(const char* category, const char* message)
{
    if(DEBUG)
        printf("INFO %s: %s", category, message);
}

void logWarning(const char* category, const char* message)
{
    if(DEBUG)
        printf("WARNING %s: %s", category, message);
}

void logError(const char* category, const char* message)
{
    if(DEBUG)
        printf("ERROR %s: %s", category, message);
}